secretmanager/snippets/create_secret.py

```diff
@@ -17,16 +17,37 @@
 """
 
 import argparse
+from typing import Optional
 
 from google.cloud import secretmanager
 
 
 # [START secretmanager_create_secret]
-def create_secret(project_id: str, secret_id: str) -> secretmanager.CreateSecretRequest:
+def create_secret(
+    project_id: str, secret_id: str, ttl: Optional[str] = None
+) -> secretmanager.Secret:
     """
     Create a new secret with the given name. A secret is a logical wrapper
     around a collection of secret versions. Secret versions hold the actual
     secret material.
+
+     Args:
+        project_id (str): The project ID where the secret is to be created.
+        secret_id (str): The ID to assign to the new secret. This ID must be unique within the project.
+        ttl (Optional[str]): An optional string that specifies the secret's time-to-live in seconds with
+                             format (e.g., "900s" for 15 minutes). If specified, the secret
+                             versions will be automatically deleted upon reaching the end of the TTL period.
+
+    Returns:
+        secretmanager.Secret: An object representing the newly created secret, containing details like the
+                              secret's name, replication settings, and optionally its TTL.
+
+    Example:
+        # Create a secret with automatic replication and no TTL
+        new_secret = create_secret("my-project", "my-new-secret")
+
+        # Create a secret with a TTL of 30 days
+        new_secret_with_ttl = create_secret("my-project", "my-timed-secret", "P30D")
```

Comment:
You either use undocumented ttl format, or have something incorrect here :) P30D doesn't match the example format you gave: `[\d]+s`

If this is some standard format for providing time, feel free to link to its documentation. Or just name it, so someone can look it up. Right now, if I wanted to have 90 days TTL, I'd just provide `7776000s` 😆

Reply:
danm) Missed that 😆 fixed